### ISSUE TITLE
Export more than just items?

### ISSUE THREAD

**Issue by crispybegs** [hay-kot/homebox#433]
### What is the problem you are trying to solve with this feature?

I only just started using homebox and spent quite a lot of time setting up locations before adding items.
When I run an export it only contains the 3 items i've added so far and none of the locations I've created (apart from those associated with the items)

Is there a way of exporting your whole home 'architecture', rather than only items?

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by hay-kot** [comment]
> Is there a way of exporting your whole home 'architecture', rather than only items?

No, not really. You can get a JSON object of the tree structure through the API, which would basically be the same data your looking for I believe. 

https://homebox.fly.dev/swagger/index.html#/Locations/get_v1_locations_tree

Exports are items only, what's the use case for exporting your entire architecture? What would you use it for?

**Comment by crispybegs** [comment]
thanks for the reply.
when i set up homebox a couple of days ago i realised i used the default config / db location in my compose, rather than my own /user/docker/homebox/data path that I'd usually use.

I'm assuming that if I change the path in the compose that whatever I've done so far will be lost? and I thought a way arounf that would be to simply export everything, change the path , then import it all again and carry on.

i am a certified idiot though, so it's possible that none of my assumptions are correct.

**Comment by icsy7867** [comment]
> thanks for the reply. when i set up homebox a couple of days ago i realised i used the default config / db location in my compose, rather than my own /user/docker/homebox/data path that I'd usually use.
> 
> I'm assuming that if I change the path in the compose that whatever I've done so far will be lost? and I thought a way arounf that would be to simply export everything, change the path , then import it all again and carry on.
> 
> i am a certified idiot though, so it's possible that none of my assumptions are correct.

With docker volumes, it will use whatever volume on the host you set it to.  If there is nothing there, then homebox initializes the data inside the folder mounted from the host.

If you are asking what I think you are asking, then yes, of you set it to a new folder that is empty it will initialize as a new deployment.  However if you copy your current files (stop the container first), let's say they are in /default/location 

Then you should be able to do a simple cp,
cp -R /default/location /user/docker/homebox/data

And use the same files there.  You obviously need to be wary of permissions and what not.  But as long as it has the same files available, it really shouldn't matter.

Alternatively you could export everything and then do an import.... But I think that is sort of what you are suggesting on the OP.

But it sounds like you just need to copy over the files to your new directory, update the compose file and rerun it.

**Comment by rpavlik** [comment]
It's hard to delete items, I'm not sure the delete button is even hooked up. so I've had to export, blow away the data, and import again, and it would have been nice to have the locations able to import. (I had some items in nested locations that import didn't link)

Also, it appears as though "parent item" is not exported in the CSV.
